paper/src/main/java/com/denizenscript/denizen/paper/events/EntityKnocksbackEntityScriptEvent.java

```diff
@@ -0,0 +1,130 @@
+
+public class EntityKnocksbackEntityScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // entity knocks back entity
+    // entity knocks back <entity>
+    // <entity> knocks back entity
+    // <entity> knocks back <entity>
+    //
+    // @Regex ^on [^\s]+ knocks back [^\s]+$
+    //
+    // @Switch in:<area> to only process the event if it occurred within a specified area.
+    // @Switch with:<item> to only process the event when the item used to cause damage (in the damager's hand) is a specified item.
+    //
+    // @Plugin Paper
+    //
+    // @Cancellable true
+    //
+    // @Triggers when an entity is knocked back from the hit of another entity.
+    //
+    // @Context
+    // <context.entity> returns the EntityTag that was knocked back.
+    // <context.damager> returns the EntityTag of the one who knocked.
+    // <context.acceleration> returns the knockback applied as a vector.
+    //
+    // @Player when the damager or damaged entity is a player. Cannot be both.
+    //
+    // @NPC when the damager or damaged entity is an NPC. Cannot be both.
+    //
+    // -->
+
+    public EntityKnocksbackEntityScriptEvent() {
+        instance = this;
+    }
+
+    public static EntityKnocksbackEntityScriptEvent instance;
+
+    public EntityTag entity;
+    public EntityTag hitBy;
+    public ItemTag held;
+    public EntityKnockbackByEntityEvent event;
+
+    @Override
+    public boolean couldMatch(ScriptPath path) {
+        return path.eventArgLowerAt(1).equals("knocks") &&
+                path.eventArgLowerAt(2).equals("back");
+    }
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        String attacker = path.eventArgLowerAt(0);
+        String target = path.eventArgLowerAt(3);
+        if (!tryEntity(hitBy, attacker) || (!tryEntity(entity, target))) {
+            return false;
+        }
+        if (!runInCheck(path, entity.getLocation())) {
+            return false;
+        }
+        if (!runWithCheck(path, held)) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public String getName() {
+        return "EntityKnocksbackEntity";
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
+        if (!isDefaultDetermination(determinationObj)) {
+            String determination = determinationObj.toString();
+            if (LocationTag.matches(determination)) {
+                event.getAcceleration().copy((LocationTag.valueOf(determination)).toVector());
+                return true;
+            }
+        }
+        return super.applyDetermination(path, determinationObj);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(
+                hitBy.isPlayer() ? hitBy.getDenizenPlayer() :
+                        (entity.isPlayer() ? entity.getDenizenPlayer() : null),
+

```

Comment:
Stray doublenewline ^
Also the newlines on the individual parts

I'm confused why this isn't formatted exactly the same as the EntityDamaged event, which I think is what you were referencing anyway?
```java
        return new BukkitScriptEntryData(damager != null && damager.isPlayer() ? damager.getDenizenPlayer() : entity.isPlayer() ? entity.getDenizenPlayer() : null,
                damager != null && damager.isCitizensNPC() ? damager.getDenizenNPC() : entity.isCitizensNPC() ? entity.getDenizenNPC() : null);
```